From 50aa7356e3c750fadb6665508e66d2f6923c7529 Mon Sep 17 00:00:00 2001
From: Yorick Peterse <yorickpeterse@gmail.com>
Date: Sat, 28 Mar 2015 14:17:27 +0100
Subject: [PATCH] "this" can not be null

Per clang 3.6 checking if "this" is null or not results in a compiler
error.
---
 vm/oop.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vm/oop.hpp b/vm/oop.hpp
index 1c55b71..8298f04 100644
--- a/vm/oop.hpp
+++ b/vm/oop.hpp
@@ -610,7 +610,7 @@ Object* const cUndef = reinterpret_cast<Object*>(0x22L);
     }
 
     void validate() const {
-      assert(this && (!reference_p() || (type_id() > InvalidType && type_id() < LastObjectType)));
+      assert(!reference_p() || (type_id() > InvalidType && type_id() < LastObjectType));
     }
 
     friend class TypeInfo;
-- 
2.3.5

